gitbutler-ui/src/lib/backend/aiProviders.ts

```diff
@@ -17,14 +20,73 @@ export interface AIProvider {
 export class ButlerAIProvider implements AIProvider {
 	constructor(
 		private cloud: ReturnType<typeof getCloudApiClient>,
-		private user: User
+		private user: User,
+		private modelKind: ModelKind
 	) {}
 
 	async evaluate(prompt: string) {
-		const messages: ButlerPromptMessage[] = [{ role: ButlerMessageRole.User, content: prompt }];
+		const messages: PromptMessage[] = [{ role: MessageRole.User, content: prompt }];
 
-		const response = await this.cloud.ai.evaluatePrompt(this.user.access_token, { messages });
+		const response = await this.cloud.ai.evaluatePrompt(this.user.access_token, {
+			messages,
+			max_tokens: 400
+		});
 
 		return response.message;
 	}
 }
+
+export class OpenAIProvider implements AIProvider {
+	private openAI: OpenAI;
+
+	constructor(
+		apiKey: string,
+		private model: OpenAIModel
+	) {
+		this.openAI = new OpenAI({ apiKey, dangerouslyAllowBrowser: true });
```

Comment:
I think it would be good to avoid calling constructors inside constructors, it will make it easier to test this code in the future. What if we made the OpenAI client a parameter?